### PR TITLE
Add Replacer to FileInfo

### PR DESCRIPTION
**Pull request by stefanpenner**
This enables a blueprint to augment the output of a given file provided.

A concrete example would be a package.json replacer.

```js
function replacer(content) {
  let pkg = JSON.parse(content);

  delete pkg.devDependencies['ember-resolver'];
  delete pkg.devDependencies['ember-ajax'];

  pkg.devDependencies['ember-strict-resolver'] = '^0.0.3';
  pkg.devDependencies['ember-native-dom-helpers'] = '^0.4.1';
  pkg.devDependencies['ember-native-dom-event-dispatcher'] = '^0.6.1';
  pkg.devDependencies['ember-fetch'] = '^2.1.0';

  return JSON.stringify(pkg, null, 2);
}


let options = { 
  /* other valid options */,
  replacer
};

let info = new FileInfo(options);
```

### CODE REVIEW

**Comment by stefanpenner** [comment]
update to blrpnt https://github.com/ember-cli/blprnt/pull/31

File: blueprints/addon/index.js [hunk 1]
```diff
@@ -60,7 +60,32 @@ module.exports = {
     contents['ember-addon'] = contents['ember-addon'] || {};
     contents['ember-addon'].configPath = 'tests/dummy/config';
 
-    this._writeContentsToFile(sortPackageJson(contents), 'package.json');
+    return JSON.stringify(sortPackageJson(contents), null, 2);
+  },
+
+  replacers: {
```
Comment (rwjblue) [inline blueprints/addon/index.js]: Maybe make this not a public API yet?

File: blueprints/addon/index.js [hunk 2]
```diff
@@ -25,12 +32,12 @@ module.exports = {
     'testem.json',
   ],
 
-  generatePackageJson() {
-    let contents = this._readContentsFromFile('package.json');
+  updatePackgeJson(content) {
```
Comment (Turbo87) [inline blueprints/addon/index.js]: there is an `a` missing 😉

Comment (stefanpenner) [inline blueprints/addon/index.js]: https://github.com/ember-cli/ember-cli/pull/7150

**Pull request merged**